api-guidelines/v4.0/API-DESIGN-GUIDELINES/POST-REQUESTS.md

```diff
@@ -0,0 +1,83 @@
+# POST Requests
+
+An HTTP POST creates an individual subordinate resource. POST is a required verb
+for non-read-only Resources.
+
+## Request URLs
+
+The URL _must_ end with the resource type. It _must not_ support provision of a
+specific identifier or query string parameters.
+
+## Request Headers
+
+There are no specific request headers associated with a `POST` request in an
+Ed-Fi API.
+
+## Request Body
+
+The request body _must_ contain exactly one item to be saved in the API
+application. Performing a `POST` with identical natural keys to a resource
+already in the data store _must_ perform an update rather than create a new
+resource (colloquially known as an "upsert"). A POST operation _must not_ allow
+a desired unique identifier to be provided to the REST API.
+
+See [Data Strictness](./DATA-STRICTNESS.md) for more information about
+validation of the request body.
+
+## Response Body
+
+The response _could_ contain the item that has just been created, optionally
+including the same [metadata](./GET-REQUESTS.md#metadata) used in a `GET`
+response.
+
+## Response Headers
+
+If successful, the URL to the new resource is returned in the `Location` HTTP
+header of the response. Example:
+
+```none
+Location: https://api.example.com/ed-fi/students/[identifier]
+```
+
+If the application supports [ETags](./REST-API.md#etags), then the ETag _must_
+be provided in an `Etag` header. Example:
```

Comment:
worth mentioning `If-None-Match` and just saying it _could_ be implemented? Currently the ODS doesn't but I know it's come up as a discussion point before.